src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -188,6 +190,24 @@ export function DashboardApp({
     getScreenshotContext,
   ]);
 
+  useMemo(() => {
```

Comment:
Can you explain why you chose `useMemo` over `useEffect`?

Reply:
I was trying to useEffect initially but the useEffect was affecting the app state updates with the startSyncingDashboardUrlState() in its own useEffect. With useMemo, this is still allowing the proper maximizing and minimizing behavior but not impacting the app state as we would expect.

Reply:
Hmmm. `useMemo` and `useEffect` should fire the same way based on the dependency array - so if a `useEffect` and `useMemo` share a dependency array, they will both be triggered when one of those things changes. So I'm not sure I'm convinced that switching to a `useMemo` is actually... fixing anything? 🤔 Can you explain the differences you were seeing? 

To me, this is hinting that something else is going on - but it's possible I'm misunderstanding 🙇

Reply:
@rshen91 I think this was missed